### ISSUE TITLE
V2: cli not finding require'd modules

### ISSUE THREAD

**Issue by faithhack** [jscad/OpenJSCAD.org#504]
I can't seem to use the same code in the cli and web when requiring a local module. 
When I require a module in a sub directory and render with a local web instance, it works.
I expect webRequire.js is saving me from doing 'bad things' with untrusted code over the web, but cli.js should just trust my includes, shouldn't it.
Using a full path in the require() returns the same error but with the full path in the error message with '/home/...' and not 'home/...' when using cli.js.

## Expected Behavior
`const { extrudeText } = require('./fonts/index.js')`
makes exported function in file ./fonts/index.js available when using localhost web app and cli
## Actual Behavior
gets error whe using cli:
`Error: No file home/chris/Projects/3DPrinting/MiscDesigns/fonts/index.js found`
`    at _require (/home/chris/lib/JSCAD-dev/OpenJSCAD.org/packages/core/code-loading`/webRequire.js:149:13)
notice file path is expanded to be relative as  'home/..'
## Steps to Reproduce the Problem

  1. create module in subdir in design project
  2. drag and drop to web app - it works
  3. run it with cli.js - gets the above error

## Specifications

  - Version: commit 5abe983 (HEAD -> V2, origin/V2)
  - Platform: Linux (ubuntu 19.10)
  - Environment:  (firefox 72.0.1, localhost server, nodejs v8.16.1)

**Comment by kaosat-dev** [comment]
Thanks for the report @faithhack ! Looking into this asap.
One bit of extra info that might help: could you please tell me if the 'root' folder for your design is **MiscDesigns** in the example above, or some other folder ?

**Comment by faithhack** [comment]
Yes, `MiscDesigns` is my root folder for the design. I get the same error if I run the cli from this folder: 
`node /path-to-jscad/packages/cli/cli.js ./mypart.js`
and if I run from the cli directory and use the absolute path to mypart.js:
`node ./cli.js /home/chris/Projects/3DPrinting/MiscDesigns/mypart.js`
Have also made sure the package.json `main` point to mypart.js

**Comment by faithhack** [comment]
Was just trying to get more info and found that requiring a file in my current directory (root of my design) also give a similar error when using the cli, so it's to related to accessing the subdirectory.

**Comment by faithhack** [comment]
Maybe another clue:
Passing my MiscDesigns **folder** to cli instead of the individual file works if I symlink OpenJSCAD/packages to @jscad
relative and full path require()s work, calling modules in node_modules work as expected.

This tanks drag-n-drop to the web app in Firefox, though, or just drives my load average though the roof.
Remove the symlink to @jscad and firefox/webapp works correctly. (but then cli fails)

**Comment by kaosat-dev** [comment]
ahhh thanks for the details @faithhack , now I see the issue , doh' ! 
You can only use require() on a ***folder*** not a single file/module indeed. 
For security reasons and to have things coherent between CLI & web use (drag & droping of a folder) So you can always go into nested folders but never higher up.

However if you run into any issues with the use of folders, that does indeed seem like a bug

> if I symlink OpenJSCAD/packages to node_modules/@jscad.
which nodes_modules folder is that ? the one from jscad or one inside your design ?

> This tanks drag-n-drop to the web app in Firefox, though, or just drives my load average though the roof. 

Ok this is really weird, we even have unit tests specifically to cover hybrid CLI/web use on a folder.
I did find some weird dependency issue (internal one). Will push a fix tomorrow

**Comment by z3dev** [comment]
NODE project:
- mymodule.js
- MiscDesigns
  - mymodule.js
  - fonts
    - index.js

**Comment by z3dev** [comment]
JSCAD project:
- mypart.js
- MiscDesigns
  - mypart.js
  - fonts
    - index.js

mypart.js
```
const {primitives} = require('@jscad/modeling')
const func1 = require('./MiscDesigns/fonts/index.js')

const main = () => {
  return primitives.sphere({radius: func1(), segments: 72})
}
module.exports = {main}
```

**Comment by kaosat-dev** [comment]
@z3dev this was on purpose to be honest : yes it works by using the node require() (which has full access to the file system), but in our case, it works the same was as it does in the browser: you need to pass a full folder (or set of files, but you lose the directory structure) in order for require() to be able to access them

**Comment by z3dev** [comment]
@faithhack the latest V2 branch fixes this issue. please try, and if fine then close this issue.

however, we strongly suggest working with 'project' folders. as @kaosat-dev mentioned, the use of a folder will allow the same project to be used for CLI, APP or WEB.

**Comment by z3dev** [comment]
Fixed via #583
